Add split methods.

lrhn opened the pull request:
Since this was considered a feature not supported by the library, and it was no problem adding it.

You could already do something manually similar using the `CharacterRange` iterator, but for end users, that's probably going to be a little too convoluted.

lrhn committed:
Add split methods.

mit-mit commented on lib/src/characters.dart:
```diff
@@ -186,6 +186,40 @@ abstract class Characters implements Iterable<String> {
+  ///
+  /// Finds each occurrence of [pattern], which does not overlap with
+  /// a previously found occurrence, then the non-matched characters
+  /// before, after and between the matches are provided in first-to-last
```
s/after and/after, and/

mit-mit commented on lib/src/characters.dart:
```diff
@@ -186,6 +186,40 @@ abstract class Characters implements Iterable<String> {
+  /// only the first `maxParts - 1` occurrences of [pattern] are found
+  /// and split at.
+  /// Any further occurrences will be included in the last part.
+  /// Example:
```
Add an empty line before the example

mit-mit reviewed (approved):
lgtm with nit comments

lrhn committed:
Address comments.

Update to use neweset package: pedantic with exceptions (to not actually change anything).

InMatrix commented:
When will this PR land? It seems to be a useful method to add.

mit-mit commented:
cc @lrhn
